test/e2e/l7networkpolicy_test.go

```diff
@@ -289,6 +293,77 @@ func testL7NetworkPolicyHTTP(t *testing.T, data *TestData) {
 	})
 }
 
+func testL7NetworkPolicyHTTPLargeResponse(t *testing.T, data *TestData) {
+	clientPodName := "test-l7-http-client-selected"
+	clientPodLabels := map[string]string{"test-l7-http-e2e": "client"}
+
+	// Create a client Pod on a Node, with the Pod being selected by the test L7 NetworkPolicy as target.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithLabels(clientPodLabels).Create(data))
+	_, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", clientPodName)
+
+	// Create a hostNetwork server Pod on another Node, with the Pod being selected by the test NetworkPolicy as the
+	// destination.
+	serverPodName := "test-l7-http-server"
+	serverPodLabels := map[string]string{"test-l7-http-e2e": "server"}
+	cmd := []string{"/agnhost", "netexec", "--http-port=8081"}
+	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).
+		WithHostNetwork(true).
+		OnNode(nodeName(1)).
+		WithCommand(cmd).
+		WithLabels(serverPodLabels).
+		Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
+	require.NoError(t, err, "Expected IP for Pod '%s'", serverPodName)
+	serverIPs := podIPs.AsSlice()
+
+	l7ProtocolAllowsPathShell := []crdv1beta1.L7Protocol{
+		{
+			HTTP: &crdv1beta1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/shell*",
+			},
+		},
+	}
+	policyAllowPathShellName := "test-l7-http-allow-path-shell"
+	createL7NetworkPolicy(t, data, true, policyAllowPathShellName, 1, clientPodLabels, serverPodLabels, ProtocolTCP, p8081, l7ProtocolAllowsPathShell)
+	time.Sleep(networkPolicyDelay)
+
+	for _, ip := range serverIPs {
+		baseURL := net.JoinHostPort(ip.String(), strconv.Itoa(int(p8081)))
+		// Verify that the test L7 NetworkPolicy allows access to the "/shell" path with large body payload.
+		assert.Eventually(t, func() bool {
```

Comment:
In case of assertion failure, it may be a bit difficult to determine what failed, because we never log errors.
This is why we use `wait.PollUntilContextTimeout` in `checkL7LoggingResult` instead of `assert.Eventually`. An alternative is to use `assert.EventuallyWithT` with the following pattern in the condition function:
```go
if !assert.NoError(t, err) {
    return
}
```

Reply:
Done